python/ray/data/_internal/arrow_block.py

```diff
@@ -633,13 +621,13 @@ def aggregate_combined_blocks(
         """
 
         stats = BlockExecStats.builder()
+        keys = sort_key.get_columns()
 
-        keys = key if isinstance(key, list) else [key]
-        key_fn = (
-            (lambda r: tuple(r[r._row.schema.names[: len(keys)]]))
-            if key is not None
-            else (lambda r: (0,))
-        )
+        def key_fn(r):
+            if sort_key is not None:
+                return tuple(r[k] for k in keys if k in r)
```

Comment:
This is just a refactor, we're not actually doing any None comparisons. In fact here the main thing is just improving the readability (previously we'd rely on the ordering of the names, which is really odd)

Reply:
On a second thought though, this filtering is incorrect -- if i have key as (A, B) where A has null value this will produce tuple as just `(b)` which is incorrect